core/services/relay/evm/codec/byte_string_modifier.go

```diff
@@ -0,0 +1,49 @@
+package codec
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+// EVMAddressModifier implements the AddressModifier interface for Ethereum addresses.
+// It handles encoding and decoding Ethereum addresses with EIP-55 checksums and hex encoding.
+type EVMAddressModifier struct{}
+
+func (e EVMAddressModifier) EncodeAddress(bytes []byte) (string, error) {
+	if len(bytes) != e.Length() {
+		return "", fmt.Errorf("%w: got length %d, expected 20 for bytes %x", commontypes.ErrInvalidType, len(bytes), bytes)
+	}
+	// Convert bytes to common.Address type and return the EIP-55 encoded string.
+
+	return common.BytesToAddress(bytes).Hex(), nil
+}
+
+// DecodeAddress takes an EIP-55 encoded Ethereum address (e.g., "0x...") and decodes it to a 20-byte array.
+func (e EVMAddressModifier) DecodeAddress(str string) ([]byte, error) {
+	// Remove the "0x" prefix if present.
```

Comment:
nits, but most of these comments are self explanatory from code, we should use comments, comment the why, not the what